pype/plugins/global/publish/integrate_new.py

```diff
@@ -764,3 +799,141 @@ def template_name_from_instance(self, instance):
             ).format(family, task_name, template_name))
 
         return template_name
+
+
+    def get_rootless_path(self, anatomy, path):
+        """  Returns, if possible, path without absolute portion from host
+             (eg. 'c:\' or '/opt/..')
+             This information is host dependent and shouldn't be captured.
+             Example:
+                 'c:/projects/MyProject1/Assets/publish...' >
+                 '{root}/MyProject1/Assets...'
+
+        Args:
+                anatomy: anatomy part from instance
+                path: path (absolute)
+        Returns:
+                path: modified path if possible, or unmodified path
+                + warning logged
+        """
+        success, rootless_path = (
+            anatomy.find_root_template_from_path(path)
+        )
+        if success:
+            path = rootless_path
+        else:
+            self.log.warning((
```

Comment:
Question came from my mind: Should we raise exception if rootless path was not found? Because in that case the file can't be used in representation files for sites, right?

Reply:
Right, but shouldn't be this stopped by some validator prior Integration phase?